test/blackbox-tests/test-cases/vendor/run.t

```diff
@@ -40,3 +40,45 @@ The same directory cannot be marked as both vendored and data-only
   Error: Directory dir was marked as vendored and data_only, it can't be marked
   as both.
   [1]
+
+The current directory cannot be marked as vendored
+
+  $ dune build --root self-vendored
+  Entering directory 'self-vendored'
+  File "dune", line 1, characters 15-16:
+  1 | (vendored_dirs .)
+                     ^
+  Error: Invalid sub-directory name "."
+  Hint: did you mean (vendored_dirs *)?
+  [1]
+
+The current directory cannot be marked as data-only
+
+  $ dune build --root self-data-only
+  Entering directory 'self-data-only'
+  File "dune", line 1, characters 16-17:
+  1 | (data_only_dirs .)
+                      ^
+  Error: Invalid sub-directory name "."
+  Hint: did you mean (data_only *)?
+  [1]
+
+Only direct subdirectories can be marked as vendored
+
+  $ dune build --root deep-subfolder-vendor
+  Entering directory 'deep-subfolder-vendor'
+  File "dune", line 1, characters 15-18:
+  1 | (vendored_dirs a/b)
+                     ^^^
+  Error: Invalid sub-directory name "a/b"
+  [1]
+
+Only direct subdirectories can be marked as data-only
+
+  $ dune build --root deep-subfolder-dataonly
+  Entering directory 'deep-subfolder-dataonly'
+  File "dune", line 1, characters 16-19:
+  1 | (data_only_dirs a/b)
+                      ^^^
+  Error: Invalid sub-directory name "a/b"
```

Comment:
I think the error message can be improved in this case. Something like:

```
Error: Only immediate sub-directories may be specified. To ignore "a/b", write "(data_only_dirs b)" in a/dune
```

Reply:
Good idea, it should work with 3b1be6c